### PR TITLE
Make location displayed by dd() identical to that by debug().

### PR DESCRIPTION
**Pull request by ADmad**
Refs #9989

### CODE REVIEW

File: src/basics.php [hunk 1]
```diff
@@ -51,7 +51,10 @@ function debug($var, $showHtml = null, $showFrom = true)
         $line = '';
         $lineInfo = '';
         if ($showFrom) {
-            $trace = Debugger::trace(['start' => 1, 'depth' => 2, 'format' => 'array']);
+            $trace = Debugger::trace(['start' => 1, 'depth' => 3, 'format' => 'array']);
+            if (count($trace) > 1) {
```
Comment (markstory) [inline src/basics.php]: Won't this always be true if there are more than 2 stack frames?

Comment (ADmad) [inline src/basics.php]: You are right, I had tested only but putting debug() and dd() in controller and got the expected result.  But for e.g. putting debug() in a model class yields incorrect result. I'll if there's a way to properly get the stack frame needed.

Comment (markstory) [inline src/basics.php]: Could walk the stack and find the call containing debug, dd, or other functions and take the next frame?

Comment (ADmad) [inline src/basics.php]: Done. Tests are now passing and I also check manually by putting debug() and dd() at various locations.

File: src/basics.php [hunk 2]
```diff
@@ -51,7 +51,13 @@ function debug($var, $showHtml = null, $showFrom = true)
         $line = '';
         $lineInfo = '';
         if ($showFrom) {
-            $trace = Debugger::trace(['start' => 1, 'depth' => 2, 'format' => 'array']);
+            $trace = Debugger::trace(['start' => 1, 'depth' => 3, 'format' => 'array']);
+            if (count($trace) > 1
+                && $trace[1]['function'] === 'dd'
```
Comment (dereuromark) [inline src/basics.php]: Having a method check for another method (dd) which calls it inside a method (debug), that doesn't sound good.
I would then rather keep things as they are.

Comment (markstory) [inline src/basics.php]: The current implementation isn't debug() + die() though. Its pr() + die(). We could rename the dd() function as an alternative.

Comment (josegonzalez) [inline src/basics.php]: Why isn't it `debug() + die()`? We could name it `diebug()` then :)

Comment (dereuromark) [inline src/basics.php]: @markstory It is debug()+die()
See https://github.com/cakephp/cakephp/blob/master/src/basics.php#L173

Comment (markstory) [inline src/basics.php]: My mistake, I'm remembering an older version of the code.

Comment (ADmad) [inline src/basics.php]: >  Having a method check for another method (dd) which calls it inside a method (debug), that doesn't sound good.

Normally yes but this is just a debugging tool and when dealing which stack traces such specific checks are often required. Having a consistent display for developer is more important here IMO.

Only other option I see is moving the location display code into a separate function which can be used by both debug() and dd() but adding another global function would be even less ideal.

Comment (markstory) [inline src/basics.php]: It could be a method on `Debugger` instead of another global method.

**Comment by ADmad** [comment]
I added Debugger::printVar()

**Comment by dereuromark** [comment]
That looks indeed cleaner 👍

**Comment by ADmad** [comment]
@markstory Is this good to go?

File: src/basics.php [hunk 3]
```diff
@@ -170,9 +130,13 @@ function dd($var, $showHtml = null)
...
-        debug($var, $showHtml, false);
-        $backtrace = debug_backtrace(false, 1);
-        pr('dd-location: ' . $backtrace[0]['file'] . ':' . $backtrace[0]['line']);
+        $trace = Debugger::trace(['start' => 1, 'depth' => 2, 'format' => 'array']);
+        $location = [
+            'line' => $trace[0]['line'],
+            'file' => $trace[0]['file']
+        ];
+
+        Debugger::printVar($var, $location);
```
Comment (dereuromark) [inline src/basics.php]: $showHtml got indeed lost in this PR - as mentioned in https://github.com/cakephp/cakephp/pull/10304

**Pull request merged**